src/controllers/action-queue/ActionProgram.ts

```diff
@@ -1,10 +1,19 @@
-import { ActionOp, ActionProgram } from './types'
+import ENV from '../../../env.json'
+import s from '../../locales/strings'
+import { ActionOp, ActionProgram, ActionProgramCompleteMessage } from './types'
 
-export async function makeActionProgram(actionOp: ActionOp, id?: string): Promise<ActionProgram> {
-  const programId = Date.now().toString() + (id ?? '')
+// TODO: Move handling from ActionProgramUtil methods here, use this instead of
+// creating individual ActionOps via ActionProgramUtil methods on the manage scenes
+export async function makeActionProgram(actionOp: ActionOp, completeMessage?: ActionProgramCompleteMessage): Promise<ActionProgram> {
+  const programId = Date.now().toString()
 
   return {
     programId,
-    actionOp
+    actionOp,
+    completeMessage: completeMessage ?? {
+      title: s.strings.action_display_title_complete_default,
+      message: s.strings.action_display_message_complete_default
+    },
+    mockMode: ENV.ACTION_QUEUE.mockMode
```

Comment:
This field isn't necessary here because it's injected by the ActionQueueStore

Reply:
I like the redundancy regardless, for debugging prior to the ActionQueueStore or if implementation changes